Neos.EventSourcedContentRepository/Classes/Domain/Context/Workspace/WorkspaceCommandHandler.php

```diff
@@ -187,38 +180,41 @@ public function handlePublishWorkspace(PublishWorkspace $command)
             throw new WorkspaceDoesNotExist(sprintf('The workspace %s does not exist', $command->getWorkspaceName()), 1513924741);
         }
         $baseWorkspace = $this->workspaceFinder->findOneByName($workspace->getBaseWorkspaceName());
-
         if ($baseWorkspace === null) {
-            throw new BaseWorkspaceDoesNotExist(sprintf('The workspace %s (base workspace of %s) does not exist', $command->getBaseWorkspaceName(), $command->getWorkspaceName()), 1513924882);
+            throw new BaseWorkspaceDoesNotExist(sprintf('The workspace %s (base workspace of %s) does not exist', $workspace->getBaseWorkspaceName(), $command->getWorkspaceName()), 1513924882);
         }
+        // TODO this hack is currently needed in order to avoid the $workspace to return a previously cached content stream identifier
+        $currentContentStreamIdentifier = $this->workspaceFinder->getContentStreamIdentifierForWorkspace($command->getWorkspaceName());
+        $baseContentStreamIdentifier = $this->workspaceFinder->getContentStreamIdentifierForWorkspace($baseWorkspace->getWorkspaceName());
```

Comment:
huh, I do not get the hack yet - why is it a hack to load stuff?

Reply:
This should be
```
$currentContentStreamIdentifier = $workspace->getCurrentContentStreamIdentifier();
```
but due to some runtime caches this returned a stale value in the behat tests.. I added a further explanation and hopefully we can get rid of this nasty work around. But it took me quite a while to find the issue to begin with so I focussed on getting it to run first.